post.js

```diff
@@ -1,454 +1,376 @@
-      content.innerHTML = DOMPurify.sanitize(contentSource);
-      details.appendChild(content);
-      postContent.appendChild(details);
+      const details = document.createElement('details')
+      const summary = document.createElement('summary')
+      summary.classList.add('cw-summary')
+      summary.textContent = 'Sensitive Content (click to view)'
+      details.appendChild(summary)
+      const content = document.createElement('p')
+      content.innerHTML = DOMPurify.sanitize(contentSource)
+      details.appendChild(content)
+      postContent.appendChild(details)
+    } else if (summary) {
+      // Non-sensitive content with a summary (post title)
+      const details = document.createElement('details')
+      const summaryElement = document.createElement('summary')
+      summaryElement.textContent = summary // Post title goes here
+      details.appendChild(summaryElement)
+
+      // Adding the "Show more" and "Show less" toggle text
+      const toggleText = document.createElement('span')
+      toggleText.textContent = 'Show more'
+      toggleText.classList.add('see-more-toggle')
+      summaryElement.appendChild(toggleText)
+
+      const contentElement = document.createElement('p')
+      contentElement.innerHTML = DOMPurify.sanitize(jsonLdData.content)
+      details.appendChild(contentElement)
+      postContent.appendChild(details)
+
+      // Event listener to toggle the text of the Show more/Show less element
+      details.addEventListener('toggle', function () {
+        toggleText.textContent = details.open ? 'Show less' : 'Show more'
+      })
     } else {
-      const content = document.createElement("p");
-      content.innerHTML = DOMPurify.sanitize(contentSource);
-      postContent.appendChild(content);
+      const content = document.createElement('p')
+      content.innerHTML = contentDOM.body.innerHTML
+      postContent.appendChild(content)
     }
 
     // Append the content to the post container
-    postContainer.appendChild(postContent);
-
-    // Footer of the post, which will contain the full published date and platform
-    const postFooter = document.createElement("footer");
-    postFooter.classList.add("post-footer");
-    const fullDate = document.createElement("div");
-    fullDate.classList.add("full-date");
-    fullDate.textContent = formatDate(jsonLdData.published) + " · reader web";
-    postFooter.appendChild(fullDate);
+    postContainer.appendChild(postContent)
+
+    // Footer of the post, which will contain the full published date and platform, but only the date is clickable
+    const postFooter = document.createElement('footer')
+    postFooter.classList.add('post-footer')
+
+    // Create a container for the full date and additional text
+    const dateContainer = document.createElement('div')
+
+    // Create the clickable link for the date
+    const fullDateLink = document.createElement('a')
+    fullDateLink.href = `/post.html?url=${encodeURIComponent(jsonLdData.id)}`
```

Comment:
Instead of linking to the id, use this: 

```js
function fetchLink(data) {
  let {url, id} = data

  if(!url) return id
  if(typeof url === 'string') return url
  if(Array.isArray(url)) {
    const firstLink = url.find((item) => (typeof item ==='string') || item.href)
    if(firstLink) return firstLink
  } else if(url.href) {
    return url.href
  }
  return id
}
```

(might have bugs plz test first)

Reply:
Yep it works